### PR TITLE
fix reqexpFORpass and add prevalidation before user click on submit btn

### PR DESCRIPTION
**Pull request by FirstDayAtWork**

### CODE REVIEW

File: static/scripts/app-register.js [hunk 1]
```diff
@@ -1,7 +1,55 @@
...
+function keyDownEvents(){
+    const REGEXP = regExpDelivery();
+
+    for(let i = 0; i < formInputs.length; i++){
+
+        formInputs[i].addEventListener('input', () => {
+            
+            let user_err = document.createElement('small');
+            if(!REGEXP[i].test(formInputs[i].value)){
+                // console.log(formInputs[i].nextElementSibling.nodeName)
```
Comment (Lifeissimple-zxc) [inline static/scripts/app-register.js]: @FirstDayAtWork We should remove this at some point, not a blocker for now.

File: static/scripts/app-register.js [hunk 2]
```diff
@@ -1,7 +1,55 @@
...
+                    formInputs[i].nextElementSibling.remove();
+                }
+                user_err.classList.add('user-err');
+                user_err.innerText = keydownMessages(formInputs[i])[i];
+                formInputs[i].style.border = '2px solid #b90909'
+                formInputs[i].after(user_err);
+                formbtn.disabled = true
+                
+            } else if(REGEXP[i].test(formInputs[i].value)){
+                if(formInputs[i].nextElementSibling?.classList.contains('user-err')){
```
Comment (Lifeissimple-zxc) [inline static/scripts/app-register.js]: We do this within both paths of if and else, why not move it outside?

File: static/scripts/app-register.js [hunk 3]
```diff
@@ -1,7 +1,55 @@
...
+        })
+    }
+    
+    
+}
+
+keyDownEvents()
+
+function keydownMessages(inputName){
+    let arr = [];
```
Comment (Lifeissimple-zxc) [inline static/scripts/app-register.js]: Why is it an array and not a map (object) where keys are names of the elements?  Like
`{
"username": "message for username"
}`

File: static/scripts/app-register.js [hunk 4]
```diff
@@ -123,4 +145,35 @@ formbtn.addEventListener('click', async (e) => {
...
+            // [a-z]{2,3}$ - It should contain two or three alphabetical characters at the end of the string. The ‘$’ represents the end of the string.
+           
+        const REGPASS = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*]).{8,}$/;
+            // old one : /^(?=.*[A-Z])(?=.*[0-9])(?=.*[a-z])[a-zA-Z0-9]{6,}$/
+            // At least 8 characters long
+            // contains a lowercase letter
+            // contains an uppercase letter
+            // contains a digit
+            // containes a special character
+        let arr = [REGUSER, REGMAIL, REGPASS];
```
Comment (Lifeissimple-zxc) [inline static/scripts/app-register.js]: Same, I think it should be a object

File: static/scripts/app-register.js (hunk 2 above)
Comment (FirstDayAtWork) [reply static/scripts/app-register.js]: quickfix

fixed

> We do this within both paths of if and else, why not move it outside?

File: static/scripts/app-register.js (hunk 3 above)
Comment (FirstDayAtWork) [reply static/scripts/app-register.js]: coz it's simpler for me to do this kind of stuff with Arrays []

> Why is it an array and not a map (object) where keys are names of the elements? Like `{ "username": "message for username" }`

File: static/scripts/app-register.js (hunk 4 above)
Comment (FirstDayAtWork) [reply static/scripts/app-register.js]: > Same, I think it should be a object

i can rewrite this with Map but should i ?

File: static/scripts/app-register.js (hunk 1 above)
Comment (FirstDayAtWork) [reply static/scripts/app-register.js:15]: > @FirstDayAtWork We should remove this at some point, not a blocker for now.

File: static/scripts/app-register.js [hunk 5]
```diff
@@ -123,4 +145,35 @@ formbtn.addEventListener('click', async (e) => {
...
+
+        const REGMAIL = /^[a-z0-9]+@[a-z]+\.[a-z]{2,3}$/
+            //  ^ - It is the start of the string.
+            // [a-z0-9]+ - Any character between a to z and 0 to 9 at the start of the string.
+            // @ - The string should contains ‘@’ character after some alphanumeric characters.
+            // [a-z]+ - At least one character between a to z after the ‘@’ character in the string.
+            // \. – Email should contain the dot followed by some characters followed by the ‘@’ character
+            // [a-z]{2,3}$ - It should contain two or three alphabetical characters at the end of the string. The ‘$’ represents the end of the string.
+           
+        const REGPASS = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*]).{8,}$/;
```
Comment (Lifeissimple-zxc) [inline static/scripts/app-register.js]: go lib we use for hashing does not support  >72 chars, can we please account for that in REGEX?

File: static/scripts/app-register.js (hunk 2 above)
Comment (Lifeissimple-zxc) [reply static/scripts/app-register.js]: good, please commit it and I will close the comment :)

File: static/scripts/app-register.js (hunk 3 above)
Comment (Lifeissimple-zxc) [reply static/scripts/app-register.js]: Your call, I did not see this pattern much, hence the comment. You create a function that simply returns you an array and you always know the order of elements there. In addition, you know what page item maps to what item of array. To me it's a pure map use case.

Comment (FirstDayAtWork) [reply static/scripts/app-register.js]: alright alright, lemme try this w/ Map and then we'll see

> Your call, I did not see this pattern much, hence the comment. You create a function that simply returns you an array and you always know the order of elements there. In addition, you know what page item maps to what item of array. To me it's a pure map use case.

File: static/scripts/app-register.js (hunk 5 above)
Comment (FirstDayAtWork) [reply static/scripts/app-register.js]: done

> go lib we use for hashing does not support >72 chars, can we please account for that in REGEX?

**Pull request merged**